src/librustdoc/html/static/js/main.js

```diff
@@ -215,10 +241,12 @@ function hideThemeButtonState() {
             searchState.input.blur();
         },
         showResults: function(search) {
+            settingsState.hideSettings();
```

Comment:
search shouldn't need to know about hiding settings, and settings shouldn't need to know about hiding search. The fact that they do indicates we don't have good boundaries around components.

Proposal: Add a placeholder element to the HTML. Whenever we activate a JS "pseudo-page" like search or settings, put the HTML for that pseudo-page into the placeholder element. That way if we need to render search, we automatically overwrite settings, and vice versa. That can scale to any number of pseudo-pages we need.

Reply:
I think it's a good idea, just not sure how you mean it: do you mean we duplicate the DOM into this placeholder from the element we want to display? If so, won't it be pretty bad for page size and speed?

Reply:
Just thought but maybe you were saying that should move the DOM between the two wrappers? If so, I'll check tomorrow how to do so.

Reply:
Nope, not proposing to duplicate the DOM into the placeholder.

Right now, the Rust code generates HTML like this:

```
<section id="search" class="content hidden"></section>
```

And the JS appends children to `#search`. I'm proposing we rename this element - for instance `#overlay`. When search needs to render into it, search will overwrite its contents (in JS). When settings needs to render into it, settings will overwrite its contents (in JS).

Reply:
But that means we have to re-render it every time, which isn't great... Both search results and the settings menu,

Reply:
We can move DOM elements pretty easily apparently:

```js
while (oldParent.childNodes.length) {
    newParent.appendChild(oldParent.firstChild);
}
```